### PR TITLE
Fixed #20456 -- Added CBV Testing docs

### PR DESCRIPTION
**Pull request by felipe-lee**
Adding the documentation for class-based view testing per the discussion in #11908

### CODE REVIEW

**Comment by felipe-lee** [comment]
@carltongibson I've updated the docs based on your recommendations. Let me know if this is good. Sorry about the delay.

File: docs/topics/testing/advanced.txt [hunk 1]
```diff
@@ -67,6 +67,45 @@ The following is a unit test using the request factory::
...
+the view, then passing a ``request`` to ``setup()``, before proceeding
+with your test's code::
+
+    # tests.py
+    from django.test import RequestFactory, TestCase
+    from .views import HomeView
+
+
+    class HomePageTest(TestCase):
+
```
Comment (felixxm) [inline docs/topics/testing/advanced.txt]: Chop blank line :thinking:

File: docs/topics/testing/advanced.txt [hunk 2]
```diff
@@ -67,6 +67,45 @@ The following is a unit test using the request factory::
...
+        def get_context_data(self, **kwargs):
+            kwargs['environment'] = 'Production'
+            return super().get_context_data(**kwargs)
+
+You may directly test the ``get_context_data()`` method by first instantiating
+the view, then passing a ``request`` to ``setup()``, before proceeding
+with your test's code::
+
+    # tests.py
```
Comment (felixxm) [inline docs/topics/testing/advanced.txt]: Maybe
```
with your test's code:

.. code-block:: python
    :caption: tests.py
```

File: docs/topics/testing/advanced.txt [hunk 3]
```diff
@@ -67,6 +67,45 @@ The following is a unit test using the request factory::
...
+Testing class-based views
+=========================
+
+In order to test class-based views outside of the request/response cycle you
+must ensure that they are configured correctly, by calling
+:meth:`~django.views.generic.base.View.setup` after instantiation.
+
+For example, assuming the following class-based view::
+
+    # views.py
```
Comment (felixxm) [inline docs/topics/testing/advanced.txt]: Maybe
```
For example, assuming the following class-based view:

.. code-block:: python
    :caption: views.py

File: docs/topics/testing/advanced.txt [hunk 4]
```diff
@@ -67,6 +67,45 @@ The following is a unit test using the request factory::
...
+    class HomeView(TemplateView):
+        template_name = 'myapp/home.html'
+
+        def get_context_data(self, **kwargs):
+            kwargs['environment'] = 'Production'
+            return super().get_context_data(**kwargs)
+
+You may directly test the ``get_context_data()`` method by first instantiating
+the view, then passing a ``request`` to ``setup()``, before proceeding
+with your test's code::
```
Comment (felixxm) [inline docs/topics/testing/advanced.txt]: Wrap at 79 chars.

**Pull request merged**